internal/metrics/readers/metrics_test.go

```diff
@@ -29,53 +26,45 @@ func TestWatchListMetrics(t *testing.T) {
 	tests := []struct {
 		name               string
 		emptyResource      bool
+		wantMetricCount    int
 		customResouceAttrs []attribute.KeyValue
 		recordMetrics      func(ctx context.Context, meter otelmetric.Meter)
 		options            []otelprometheus.Option
-		expectedFile       string
 	}{
 		{
-			name:         "batching metrics",
-			expectedFile: "testdata/readers_happy.txt",
+			name: "batching metrics",
 			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
 				Init(meter)
 				events := []watch.Event{
 					{
 						Type: watch.Added,
 					},
-					{
-						Type: watch.Error,
-					},
 				}
 				for _, event := range events {
 					WatchEvent(ctx, event)
-					DataEntry(ctx, data.MustNewEntry(&corev1.Node{}, data.STInformer, data.CTAdd))
-					DataEntry(ctx, data.MustNewEntry(&corev1.Pod{}, data.STInformer, data.CTAdd))
 					DataEntry(ctx, data.MustNewEntry(&corev1.Pod{}, data.STInformer, data.CTUpdate))
-					DataEntry(ctx, data.MustNewEntry(&corev1.Pod{}, data.STInformer, data.CTUpdate))
-					DataEntry(ctx, data.MustNewEntry(&corev1.Pod{}, data.STInformer, data.CTDelete))
 					StaleData(ctx, data.MustNewEntry(&corev1.Pod{}, data.STInformer, data.CTDelete))
 				}
 			},
+			wantMetricCount: 4,
 		},
 		{
-			name:         "batching metrics not initialized",
-			expectedFile: "testdata/readers_nometrics.txt",
+			name: "batching metrics not initialized",
 			recordMetrics: func(ctx context.Context, meter otelmetric.Meter) {
 				WatchEvent(ctx, watch.Event{Type: watch.Added})
 				DataEntry(ctx, data.MustNewEntry(&corev1.Node{}, data.STInformer, data.CTAdd))
 				StaleData(ctx, data.MustNewEntry(&corev1.Node{}, data.STInformer, data.CTAdd))
 			},
+			wantMetricCount: 1,
 		},
 	}
 
 	for _, test := range tests {
-		log.Println("test: ", test.name)
 		ctx := context.Background()
 		registry := prometheus.NewRegistry()
 		exporter, err := otelprometheus.New(append(test.options, otelprometheus.WithRegisterer(registry))...)
 		if err != nil {
-			t.Fatalf("failed to create prometheus exporter: %v", err)
+			t.Fatalf("TestWatchListMetrics(%s) failed to create prometheus exporter: %v", test.name, err)
```

Comment:
It looks like I wrote this as if the test was using t.Run, and it should use t.Errorf + continue instead of t.Fatalf. Sorry!

Reply:
Fixed.